test_scripts/API/PerformInteraction/011_ATF_PerformInteraction_UI_PerformInteraction_WARNINGS.lua

```diff
@@ -0,0 +1,154 @@
+      :Do(function(_,_)
+        self.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality", {allowed = true, source = "GUI", device = {id = config.deviceMAC, name = ServerAddress}})
+        EXPECT_HMICALL("BasicCommunication.ActivateApp")
+        :Do(function(_,data1)
+          self.hmiConnection:SendResponse(data1.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+        end)
+      end)
+    end
+  end)
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+end
+
+function Test:Precondition_CreateInteractionChoiceSet()
+  local cid = self.mobileSession:SendRPC("CreateInteractionChoiceSet",
+  {
+    interactionChoiceSetID = 1001,
+      choiceSet = {{
+        choiceID = 1001,
+        menuName ="Choice1001",
+        vrCommands = { "Choice1001" },
+        image = { value ="icon.png", imageType ="DYNAMIC" },
+    }}
+  })
+
+  EXPECT_HMICALL("VR.AddCommand",
+  {
+    cmdID = 1001,
+    appID = self.applications[config.application1.registerAppInterfaceParams.appName],
+    type = "Choice",
+    vrCommands = {"Choice1001"}
+  })
+  :Do(function(_,data2)
+    self.hmiConnection:SendResponse(data2.id, "VR.AddCommand", "SUCCESS", {})
+  end)
+
+  EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS" })
+  EXPECT_NOTIFICATION("OnHashChange")
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local resultCodes = {"SUCCESS", "WARNINGS", "WRONG_LANGUAGE", "RETRY", "SAVED"}
+
+for i=1,#resultCodes do
+  Test["TestStep_PerformInteraction_UI_PerfInteract_WARNINGS_and_VR_PerfInteract_"..resultCodes[i]] = function(self)
+    local cor_id = self.mobileSession:SendRPC("PerformInteraction",
+    {
+      initialText = "StartPerformInteraction",
+      initialPrompt = {{ text = "Makeyourchoice", type = "TEXT" }},
+      interactionMode = "BOTH",
+      interactionChoiceSetIDList = {1001},
+      timeout = 5000
+    })
+
+    EXPECT_HMICALL("VR.PerformInteraction",
+    {
+      initialPrompt = {{ text = "Makeyourchoice", type = "TEXT" }},
+      timeout = 5000,
+      appID = self.applications[config.application1.registerAppInterfaceParams.appName]
+    })
+    :Do(function(_,data3)
+      self.hmiConnection:SendNotification("TTS.Started")
+      self.hmiConnection:SendNotification("VR.Started")
+      self.hmiConnection:SendResponse(data3.id, "VR.PerformInteraction", resultCodes[i], {choiceID = 1001})
+      self.hmiConnection:SendNotification("TTS.Stopped")
+      self.hmiConnection:SendNotification("VR.Stopped")
+    end)
+
+    EXPECT_HMICALL("UI.PerformInteraction",
+    {
+      initialText = {fieldName = "initialInteractionText", fieldText = "StartPerformInteraction" },
+      timeout = 5000,
+      appID = self.applications[config.application1.registerAppInterfaceParams.appName]
+    })
+    :Do(function(_,data4)
+      local function uiResponse()
+        self.hmiConnection:SendResponse(data4.id,"UI.PerformInteraction", "WARNINGS", {})
+        SendOnSystemContext (self, "VRSESSION")
```

Comment:
@vvvakulenko, function SendOnSystemContext is not declared, you can declare it before local function uiResponse.
As example: 
local function SendOnSystemContext(Input_SystemContext)
        self.hmiConnection:SendNotification("UI.OnSystemContext",{ appID = self.applications[config.application1.registerAppInterfaceParams.appName], systemContext = Input_SystemContext})
      end